cmd/root.go

```diff
@@ -115,48 +99,93 @@ func initConfig() {
 	if err != nil {
 		ExitWithError(err.Error())
 	}
-
-	if err := networkConfig.ReadInConfig(); err == nil {
-		miners := networkConfig.GetStringSlice("miners")
-		sharders := networkConfig.GetStringSlice("sharders")
-		if len(miners) > 0 && len(sharders) > 0 {
-			zcncore.SetNetwork(miners, sharders)
-		}
-	}
 }
 
-func checkWalletPath(cmd *cobra.Command, args []string) {
+func loadConfigs() {
+	cfgConfig = viper.New()
+	cfgNetwork = viper.New()
 	var configDir string
 	if cDir != "" {
 		configDir = cDir
 	} else {
 		configDir = getConfigDir()
 	}
 
+	// ~/.zcn/config.yaml
+	cfgConfig.AddConfigPath(configDir)
+	if &cfgFile != nil && len(cfgFile) > 0 {
+		cfgConfig.SetConfigFile(configDir + "/" + cfgFile)
+	} else {
+		cfgConfig.SetConfigFile(configDir + "/" + "config.yaml")
+	}
+
+	minSubmit = cfgConfig.GetInt("min_submit")
+	minCfm = cfgConfig.GetInt("min_confirmation")
+	CfmChainLength = cfgConfig.GetInt("confirmation_chain_length")
+
+	// ~/.zcn/network.yaml
+	cfgNetwork.AddConfigPath(configDir)
+	if &networkFile != nil && len(networkFile) > 0 {
+		cfgNetwork.SetConfigFile(configDir + "/" + networkFile)
+	} else {
+		cfgNetwork.SetConfigFile(configDir + "/" + "network.yaml")
+	}
+
+	if err := cfgConfig.ReadInConfig(); err != nil {
+		ExitWithError("Can't read config:", err)
+	}
+
+	cfgNetwork.ReadInConfig() //nolint: errcheck
+
 	// TODO: move the private key storage to the keychain or secure storage
-	var walletFilePath string
+	// ~/.zcn/wallet.json
 	if &walletFile != nil && len(walletFile) > 0 {
-		walletFilePath = configDir + "/" + walletFile
+		cfgWallet = configDir + "/" + walletFile
 	} else {
-		walletFilePath = configDir + "/wallet.json"
+		cfgWallet = configDir + "/wallet.json"
 	}
+}
+
+var zcnIsConnected bool
+var hasWallet bool
 
-	cmdName := cmd.Name()
-	shouldCreateWallet := cmdName != "version" && cmdName != "createmswallet" 
+func initCmdContext(cmd *cobra.Command, args []string) {
 
-	if !shouldCreateWallet {
-		return
+	// connect to zcn
+	if !zcnIsConnected {
+		_, ok := withoutZCNCmds[cmd]
+		if !ok {
```

Comment:
This double negation is counterintuitive, actually all CLI commands should be offline by default, they should init zcn and load wallet when necessary. Anyway, we can have InitZCN() and LoadWallet() that work the similar way you do for WithoutZCN() and WithoutWallet().

So do `rootCmd.AddCommand(InitZCN(LoadWallet(cmd-that-need-zcn-and-wallet)` for all commands that need zcn and wallet.

Reply:
Updated naming, please check the code again. 
Almost of commands ask to initialize zcncore package and load wallet.  I would suggest to `zcncore` should be initialized and `wallet` should be loaded as default.  Actually almost of them can't work as offline.

Reply:
for now, there are only 2 commands that can work as offline
- version : without zcncore, without wallet
- createmswallet: without wallet

Reply:
@cnlangzi `recoverwallet` should also have offline option?

Reply:
Actually, `createwallet` aswell

Reply:
@sculptex  `createwallet` works with `zcncore` package. the package is asked to query network to init sdk. that is why it doesn't work as offline. recoverwallet should have the same. They have to be refactored without zcncore package if they want to work as offline.

Reply:
@cnlangzi `recoverwallet` was whole reason I raised offline issue in first place - That is someone with stored seed phrase should be able to securely regenerate their wallet offline.

I created alternate repo for this purpose, https://github.com/sculptex/zcnwalletutil I suggest if it will take too much of your time then it should just be kept as totally separate repo but needs to be forked and endorsed as official 0chain repo so users will have re-assurance of its security.

Reply:
@sculptex ok. let me try to move some code out from zcncore for wallet commands

Reply:
@sculptex  added --offline on `createmswallet` and `recoverwallet`

Reply:
@sculptex   It means register to miners. so it is online feature.